src/playlists.ts

```diff
@@ -0,0 +1,68 @@
+import * as vscode from 'vscode';
+import SpotifyWebApi = require('spotify-web-api-node');
+
+export class PlaylistsProvider implements vscode.TreeDataProvider<PlaylistAndTracks> {
+
+  constructor(private spotify: SpotifyWebApi) {}
+
+  async getChildren(element?: PlaylistAndTracks): Promise<PlaylistAndTracks[] | undefined> {
+    if (element === undefined) {
+      return this.getPlaylistAndTracks(this.spotify);
+    }
+    return element.children;
+    
+  }
+  async getTreeItem(element: PlaylistAndTracks): Promise<PlaylistAndTracks> {
+    return element;
+  }
+
+  async getPlaylistAndTracks(spotify: SpotifyWebApi) {
+    try {
+      const user = await this.spotify.getMe();
+      const playlists = await this.spotify.getUserPlaylists(user.body.id);
+      return playlists.body.items.map((playlist: any) => new PlaylistAndTracks(playlist, this.spotify));
+      } catch (err: any) {
+      vscode.window.showErrorMessage(err);
+    }
+  }
+}
+
+
+export class PlaylistAndTracks extends vscode.TreeItem {
+  name: string;
+  artists: string;
+  uri: string;
+  children: Promise<PlaylistAndTracks[]> | undefined;
+  parent: PlaylistAndTracks | undefined;
+
+  constructor (public readonly data: any, spotify?: SpotifyWebApi, parent?: PlaylistAndTracks) {
+    super(data.name);
+    
+    if (spotify) {
```

Comment:
Might be worth adding a comment here explaining how `spotify` is defined for playlists and undefined for tracks. Took me a hot minute to see that.

Reply:
Ahhh yes, that is the beauty of the whole system // the logic behind how I'm able to to use one class for both playlists & tracks

Reply:
imo it would be a lot more "elegant" if we used an interface to take out the need for an if statement, but you put your sweat and blood into this code and throwing in an interface isn't going to impress any judges or make any lives better

Reply:
Don't forget about my tears 👅